sympy/discrete/recurrences.py

```diff
@@ -109,6 +109,12 @@ def linrec(coeffs, init, n):
     else:
         b += [S.Zero]*(k - len(b)) # remaining initial values default to zero
 
+
+    return b[n] if n < k else sum(u*v for u, v in zip(_linrec_coeffs(c, n), b))
```

Comment:
I think that this could be made more widely applicable by the change that I suggested in [this comment](https://github.com/sympy/sympy/pull/18595#issuecomment-583595410).